src/apps/frontend/types/async-operation.ts

```diff
@@ -0,0 +1,18 @@
+import { ServiceError, ServiceResponse } from './service-response';
+
+export enum AsyncOperationStatus {
+  Idle = 'idle',
+  Pending = 'pending',
+  Success = 'success',
+  Error = 'error',
+}
+
+export interface AsyncOperation<T> {
+  result?: T;
+  error?: ServiceError;
+  status: AsyncOperationStatus;
```

Comment:
Async operation is an upper level abstraction than Service Error. This does not sound right

Reply:
Done